### PR TITLE
Feature/exercise structure refactoring

### PR DESCRIPTION
**Pull request by janm399**
**READY**

This PR cleans up the entire exercise mess: ``MKIncompleteExercise``, ``MKExercise``, ``MKLabelledExercise``, their ``MR*`` counterparts and all that is gone. Replaced by a much tidier tagging model.

### CODE REVIEW

File: Muvr/Entities/MRManagedExerciseSession.swift [hunk 1]
```diff
@@ -1,198 +1,146 @@
...
+    /// Viz ``setClassificationHint(_:labels:)`` and ``clearClassificationHint``
+    private(set) internal var classificationHints: [MKClassificationHint] = []
+    /// The accumulated exercise rows
+    private(set) internal var exerciseWithLabels: [ExerciseRow] = []
+    
+    /// The set of estimated exercises, used when the session is in real-time mode
+    var estimated: [MKExerciseWithLabels] = []
     /// The weight predictor
     var weightPredictor: MKScalarPredictor!
+    /// The weight predictor
```
Comment (by-dam) [inline Muvr/Entities/MRManagedExerciseSession.swift]: Comment should say ``The duration predictor`` (or remove comments as we can easily guess with the variable names ``durationPredictor``)

File: Muvr/MKExerciseType+Comparable.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Foundation
+import MuvrKit
+
+extension MKExerciseType : Comparable {
+    
+}
+
+public func <(lhs: MKExerciseType, rhs: MKExerciseType) -> Bool {
+    switch (lhs, rhs) {
+    case (.ResistanceTargeted(let lmgs), .ResistanceTargeted(let rmgs)) where lmgs.count == 1 && rmgs.count == 1:
```
Comment (by-dam) [inline Muvr/MKExerciseType+Comparable.swift]: why not sorting the muscle groups then comparing the titles of their first arg?

File: Muvr/MRAppDelegate+UITesting.swift [hunk 3]
```diff
@@ -12,53 +12,41 @@ import MuvrKit
...
-        return [
-            "resistanceTargeted:arms/biceps-curl", "resistanceTargeted:arms/triceps-extension", "resistanceTargeted:shoulders/lateral-raise", "resistanceTargeted:arms/reverse-cable-curl",
-            "resistanceTargeted:legs/squat", "resistanceTargeted:legs/lunges", "resistanceTargeted:legs/dumbbell-calf-raise",
-            "resistanceTargeted:back/barbell-row", "resistanceTargeted:back/back-extensions", "resistanceTargeted:back/lateral-pulldown-straight",
-            "resistanceTargeted:core/oblique-crunches", "resistanceTargeted:core/suitcase-crunches", "resistanceTargeted:core/side-dips", "resistanceTargeted:core/crunches",
-            "resistanceTargeted:chest/dumbbell-flyes", "resistanceTargeted:chest/angle-chest-press", "resistanceTargeted:chest/butterfly-machine"
-        ]
-    }
-    
+//    ///
```
Comment (by-dam) [inline Muvr/MRAppDelegate+UITesting.swift]: can we remove commented lines ?

File: Muvr/MRExerciseLabelViews.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import Foundation
+import MuvrKit
+
+struct MRExerciseLabelViews {
+    
+    static func scalarViewForLabel(label: MKExerciseLabel, frame: CGRect) -> (UIView, MRScalarExerciseLabelSettable)? {
```
Comment (by-dam) [inline Muvr/MRExerciseLabelViews.swift]: :+1: for the label views

File: Muvr/MRManualViewController.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import UIKit
+import MuvrKit
+
+class MRGoalsViewController : UIViewController, UITableViewDataSource, UITableViewDelegate {
```
Comment (by-dam) [inline Muvr/MRManualViewController.swift]: Now the phone app really works with no watch!

**Comment by by-dam** [comment]
:+1: That's a big one but the new Exercise classes are worth it.
We are no longer uploading user's sessions to S3. 
Are we no longer interested in sessions data to improve the classifier ?

**Comment by janm399** [comment]
We can slowly bring things in from the latest develop, which I'm going to tag. We are ultimately interested in data from the users, in S3. But let's get this initial structure right and then think about that.

As for the models, they will be in-app purchases.

File: Muvr/MRExerciseLabelViews.swift (hunk 4 above)
Comment (janm399) [inline Muvr/MRExerciseLabelViews.swift]: I'm not entirely happy with this structure, but it's OK to start with.

File: MuvrKit/MKExerciseType.swift [hunk 6]
```diff
@@ -16,7 +16,25 @@ public enum MKExerciseType : Equatable, Hashable {
...
+    /// Indicates whether self is contained in that. This is useful for targeted 
+    /// resistance with multiple muscle groups, where strict equality does not work
+    /// - parameter that: the other ET
+    /// - returns: true if self is contained within that
+    ///
+    public func isContainedWithin(that: MKExerciseType) -> Bool {
+        switch (self, that) {
+        case (.ResistanceTargeted(let lmgs), .ResistanceTargeted(let rmgs)):
+            return lmgs.reduce(true) { result, lmg in
+                return rmgs.contains(lmg)
```
Comment (tmbo) [inline MuvrKit/MKExerciseType.swift]: Aren't we missing a `& result here`?

Comment (janm399) [inline MuvrKit/MKExerciseType.swift]: Yes.

**Pull request merged**